test/controllers/api/v2/domains_controller_test.rb

```diff
@@ -64,42 +64,42 @@ class Api::V2::DomainsControllerTest < ActionController::TestCase
     refute Domain.find_by_id(domain['id'])
   end
 
-  # test that taxonomy scope works for api for domains
-  def setup
-    taxonomies(:location1).domain_ids = [domains(:mydomain).id, domains(:yourdomain).id]
-    taxonomies(:organization1).domain_ids = [domains(:mydomain).id]
-  end
-
-  test "should get domains for location only" do
-    get :index, params: { :location_id => taxonomies(:location1).id }
-    assert_response :success
-    assert_equal taxonomies(:location1).domains.length, assigns(:domains).length
-    assert_equal assigns(:domains), taxonomies(:location1).domains
-  end
+  context "taxonomy scope" do
+    let (:mydomain) { domains(:mydomain) }
+    let (:yourdomain) { domains(:yourdomain) }
+    let (:loc) { FactoryBot.create(:location, domains: [mydomain, yourdomain]) }
+    let (:org) { FactoryBot.create(:organization, domains: [mydomain]) }
+
+    test "should get domains for location only" do
+      get :index, params: { :location_id => loc.id }
+      assert_response :success
+      assert_equal loc.domains.length, assigns(:domains).length
+      assert_equal assigns(:domains).sort, loc.domains.sort
```

Comment:
I beliebe assert_same_elements could help here.

Reply:
nice! didn't know that one.